analyzers/PapasFromFccsw.py

```diff
@@ -29,30 +30,48 @@ class PapasEventFromRoot(Analyzer):
     '''
 
     def __init__(self, *args, **kwargs):
-        super(PapasEventFromRoot, self).__init__(*args, **kwargs)
+        super(PapasFromFccsw, self).__init__(*args, **kwargs)
 
     def process(self, event):
         event.papasevent = PapasEvent(event.iEv)
         papasevent = event.papasevent
 
         #make a dict from the gen_particles list so that it can be stored into the papasevent collections
         gen_particles = getattr(event, self.cfg_ana.gen_particles)
-        gen_particles_collection = {x.uniqueid():x for x in gen_particles}
+        for g in gen_particles:
+            #set the papas identifiers for use in DAG
+            g.set_dagid(Identifier.make_id(Identifier.PFOBJECTTYPE.PARTICLE, g.objid()[0], 'g', g.p4().E()))
+        #make a dict from the gen_particles list so that it can be stored into the papasevent collections           
+        gen_particles_collection = {x.dagid():x for x in gen_particles}
```

Comment:
I would fill the dictionary and set the dag id in the same loop (always good to reduce the number of loops)